File: algorithm/src/test/java/programmers/level0/종이자르기/SolutionTest.java
package programmers.level0.종이자르기;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class SolutionTest {
    private Solution solution = new Solution();

    @Test
    void case1() {
        int m = 2;
        int n = 2;
        int result = 3;
        assertThat(solution.solution(m, n)).isEqualTo(result);
    }

    @Test
    void case2() {
        int m = 2;
        int n = 5;
        int result = 9;
        assertThat(solution.solution(m, n)).isEqualTo(result);
    }

    @Test
    void case3() {
        int m = 1;
        int n = 1;
        int result = 0;
        assertThat(solution.solution(m, n)).isEqualTo(result);
    }

    @Test
    void case4() {
        int m = 5;
        int n = 2;
        int result = 9;
        assertThat(solution.solution(m, n)).isEqualTo(result);
    }
}

File: algorithm/src/test/kotlin/io/yomni/programmers/level1/id12937/SolutionTest.kt
package io.yomni.programmers.level1.id12937

import io.yomni.programmers.level1.id12937.Solution
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource

@DisplayName("SolutionTest")
class SolutionTest {

    companion object {
        @JvmStatic
        fun argumentsProvider() = listOf(
            Arguments.of(3, "Odd"),
            Arguments.of(4, "Even"),
        )
    }

    @ParameterizedTest
    @MethodSource("argumentsProvider")
    fun solution(input: Int, expected: String) {
        val solution = Solution()
        assertThat(solution.solution(input)).isEqualTo(expected)
    }

}
